### PR TITLE
V0.3.3

### PR DESCRIPTION
**Pull request by orionarcher**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by orionarcher** [comment]
@conda-forge-admin, please rerender

**Comment by orionarcher** [comment]
@hmacdope trying again here

**Comment by orionarcher** [comment]
For this version I tried to modify the `meta.yml` in `solvation_analysis` to include the missing dependencies but it didn't seem to change anything.

aaaaah! I'm really not sure where `pathlib`, `rdkit`, and `pytest` are being left out. I'm particuarly confused by the missing `pathlib` since it is part of the core library. @hmacdope you mentioned I'd need to make some changes to `pyproject.toml`, could you be a bit more specific about what those are? The "missing" dependencies are already included there.

**Comment by orionarcher** [comment]
So I fixed this by adding to the list of commands `pip install rdkit pytest pathlib`. I don't know if that's a real solution though because it's purely a fix for the testing environment. That said, the problem might be some weird interaction of the test env and the dependencies that wouldn't happen on a users device.

Will that cause issues once the package is actually released? What should next steps be?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,35 +1,45 @@
...
     - scipy
+    - plotly
+    - statsmodels
+    - rdkit
 
 test:
   imports:
     - solvation_analysis
   commands:
+    - pip install rdkit pytest pathlib
```
Comment (hmacdope) [inline recipe/meta.yaml]: @IAlibay this should not be necessary but I am completely stumped as to what we are doing wrong, seems to me like we should be packaging everything already. Any help would be greatly appreciated.

**Comment by hmacdope** [comment]
@orionarcher sorry for the radio silence been super busy, I am just going to try a few things and also ping some people that know more conda-forge than me.

**Comment by hmacdope** [comment]
We'll get it sorted.

**Review by hmacdope — COMMENTED** [review]
Lets ask @IAlibay if he has any ideas, i'm stumped.

File: recipe/meta.yaml (hunk 1 above)
Comment (IAlibay) [reply recipe/meta.yaml]: Yeah this seems like something went very wrong / defeats the purpose of package managing. I'll try to have a poke at things later today or tomorrow.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,35 +1,45 @@
...
   number: 0
 
 requirements:
+  build:
+    - versioneer-518
   host:
-    - python >=3.7
+    - python >=3.8
     - pip
+    - pathlib
```
Comment (IAlibay) [inline recipe/meta.yaml]: I'll be honest, I'm very confused as to why you even have pathlib present. It's a core part of the Python library, you shouldn't need it externally. You are aware that by defining it in the pyproject.toml you're actually picking up the **old** pathlib from the Python 3.2 days right?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,35 +1,45 @@
...
+    - pytest
+    - rdkit
```
Comment (IAlibay) [inline recipe/meta.yaml]: These aren't build time dependencies, they _shouldn't_ need to be here?

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,35 +1,45 @@
...
   run:
-    - python >=3.7
-    - numpy >=1.16.0
+    - python >=3.8
+    - numpy >=1.18.0
     - mdanalysis >=2.0.0
     - pandas
     - matplotlib-base
     - scipy
+    - plotly
```
Comment (IAlibay) [inline recipe/meta.yaml:33]: try adding pytest and pathlib here to see if it fixes things?

(if it does, I'd cut a 0.3.3 release and get rid of pathlib)

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,35 +1,45 @@
...
     - pip
+    - pathlib
+    - pytest
+    - rdkit
+    - versioneer-518
   run:
-    - python >=3.7
-    - numpy >=1.16.0
+    - python >=3.8
+    - numpy >=1.18.0
```
Comment (IAlibay) [inline recipe/meta.yaml]:
```suggestion
    - numpy >=1.20.0
```
This doesn't match your pyproject.toml

**Comment by orionarcher** [comment]
Thanks for the guidance. I'll implement your suggestions tomorrow and hope it fixes things.

Is there a way to test the affect of changes to `pyproject.toml` on the conda-forge build without minting a new release?

File: recipe/meta.yaml (hunk 2 above)
Comment (orionarcher) [reply recipe/meta.yaml]: I was not aware! This should definitely be removed.

**Comment by IAlibay** [comment]
> Thanks for the guidance. I'll implement your suggestions tomorrow and hope it fixes things.
> 
> Is there a way to test the affect of changes to `pyproject.toml` on the conda-forge build without minting a new release?

Probably the cleanest way is to do a pre-release build. Point the CF build to a specific git rev and you'll be able to test a package build against that.

**Comment by orionarcher** [comment]
The build is now successfully finding pytest and pathlib but it's still failing to find rdkit. It's included in `pyproject.toml`, `setup.py` and `requirements.txt` so I am unclear on why that might be or what next steps should be.

**Comment by hmacdope** [comment]
@conda-forge-admin, please rerender

**Comment by hmacdope** [comment]
@conda-forge/help-python we are a bit stuck here, `pip check` is complaining no `rdkit` but we clearly require it everywhere I can see. Any help would be super appreciated.

**Comment by beenje** [comment]
> @conda-forge/help-python we are a bit stuck here, `pip check` is complaining no `rdkit` but we clearly require it everywhere I can see. Any help would be super appreciated.

Looking at the `rdkit` conda package, there is no `dist-info` directory, so `pip` can't detect it.
There is only `site-packages/rdkit` created.

If you `pip install rdkit`, you get the following directories under `site-packages`:

```
rdkit
rdkit-2022.9.5.dist-info
rdkit.libs
```

I found this issue https://github.com/conda-forge/rdkit-feedstock/issues/104

As the wheel includes a `dist-info`, probably something to fix in the conda feedstock.

Anyway, I'd just remove `pip check` test from your recipe.

**Comment by orionarcher** [comment]
Thanks for the advice @beenje!

I think this is good to merge now. @hmacdope I'll mint a 0.3.3 (non-alpha) release on PyPI, update here, and merge if that sounds good to you?

**Comment by hmacdope** [comment]
@orionarcher sounds perfect.

**Pull request merged**